apps/conjugate_gradient/main.c

```diff
@@ -0,0 +1,217 @@
+// Copyright 2022 ETH Zurich and University of Bologna.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Author: Chi Zhang, ETH Zurich <chizhang@iis.ee.ethz.ch>
+
+
+#include <stdint.h>
+#include <string.h>
+
+#include "runtime.h"
+#include "util.h"
+#include "shared_kernel/gemv.h"
+#include "shared_kernel/spmv.h"
+#include "shared_kernel/fdotproduct.h"
+
+#ifndef SPIKE
+#include "printf.h"
+#else
+#include <stdio.h>
+#endif
+
+#define USE_SPMV 1
+#define MIN_LOSS 0.0005
+#define abs(x) (x<0? -x:x)
+
+extern uint64_t size;
+extern uint64_t step;
+extern double sparsity;
+
+extern double A[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double Ax[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double x[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double b[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double r[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double p[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double Ap[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern int32_t A_PROW[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern int32_t A_IDX[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+extern double A_DATA[] __attribute__((aligned(4 * NR_LANES), section(".l2")));
+
+void daxpy(double * x, double a, double * y, double * dest, uint64_t len){
+	#define MAX_AVL 128
+	while(len>MAX_AVL){
```

Comment:
Have you passed this through `clang-format`? 
To do so, just run `make -C apps format`